src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +265,116 @@ namespace AppInstaller::CLI::Workflow
+            AICLI_LOG(CLI, Error, << "Store client is blocked by policy. MSStore install failed.");
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_BLOCKED_BY_POLICY);
+        }
+
+        if (!installManager.GetIsAppAllowedToInstallAsync(productId).get())
+        {
+            context.Reporter.Error() << Resource::String::MSStoreInstallAppBlocked << std::endl;
+            AICLI_LOG(CLI, Error, << "App is blocked by policy. MSStore install failed. ProductId: " << Utility::ConvertToUTF8(productId));
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_APP_BLOCKED_BY_POLICY);
+        }
+
+        // Verifying/Acquiring product ownership
+        context.Reporter.Info() << Resource::String::MSStoreInstallTryGetEntitlement << std::endl;
+        GetEntitlementResult enr = installManager.GetFreeUserEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
+
+        if (enr.Status() == GetEntitlementStatus::Succeeded)
+        {
+            context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
+            AICLI_LOG(CLI, Error, << "Get entitlement succeeded.");
+        }
+        else
+        {
+            if (enr.Status() == GetEntitlementStatus::NoStoreAccount)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNoStoreAccount << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. No Store account.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::NetworkError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNetworkError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. Network error.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::ServerError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementServerError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement succeeded. Server error. ProductId: " << Utility::ConvertToUTF8(productId));
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_INSTALL_FAILED);
+        }
+
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        IVectorView<AppInstallItem> installItems = installManager.StartProductInstallAsync(
+            productId,              // ProductId
+            winrt::hstring(),       // CatalogId
+            winrt::hstring(),       // FlightId
+            L"WinGetCli",           // ClientId
+            false,                  // repair
+            false,
+            winrt::hstring(),
+            nullptr).get();
+
+        // We install one package at a time so the AppInstallItem list returned should contain only one entry.
+        auto installItem = installItems.GetAt(0);

```

Comment:
>auto installItem = installItems.GetAt(0); [](start = 8, length = 41)

We should probably log a bunch of info about these requests so that we can easily associate them with store/deployment logs.  Store request ids, PFNs, anything else that will make it easier to link the two.

Reply:
Looking at the properties of AppInstallItem, only ProductId and PFN are related to our diagnosis. Logged these 2.

---
In reply to: [455243165](https://github.com/microsoft/winget-cli/pull/493#discussion_r455243165) [](ancestors = 455243165)